### ISSUE TITLE
NullPointerException from OQueryOperatorIn.executeIndexQuery

### ISSUE THREAD

**Issue by christinedraper** [orientechnologies/orientdb#7287]
## Expected behavior  
Uses index to get result  

## Actual behavior  
Null pointer exception on server accessing index, scans cluster instead.  

## Steps to reproduce  

Create a new database with an index:
```
create class V_Entity extends V;
create property V_Entity.entityId STRING;
create index entity_id_index ON V_Entity (entityId) UNIQUE;
```
Create a vertex :
```
create vertex V_Entity set entityId="user1"
```

Run following query. It's an excerpt of a more complex query that I can post if you think the query is problematic. I know this one looks odd, it's just the simplest reproduce:

```
select from V_Entity 
    let $c = (select entityId from V_Entity 
        where entityId="user1")
    where entityId in $c 
```

The query successfully returns the expected result, but on the server I get:
```
$ANSI{green {db=test2}} Error on using index entity_id_index in query 'SELECT FROM V_Entity LET $c = (SELECT entityId FROM V_Entity WHERE entityId = "user1") WHERE entityId IN $c'. Probably you need to rebuild indexes. Now executing query using cluster scan
java.lang.NullPointerException
	at com.orientechnologies.orient.core.sql.operator.OQueryOperatorIn.executeIndexQuery(OQueryOperatorIn.java:92)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.searchForIndexes(OCommandExecutorSQLSelect.java:2184)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.searchInClasses(OCommandExecutorSQLSelect.java:1001)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLResultsetAbstract.assignTarget(OCommandExecutorSQLResultsetAbstract.java:209)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.assignTarget(OCommandExecutorSQLSelect.java:530)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.executeSearch(OCommandExecutorSQLSelect.java:512)
<snip>
```

**Comment by luigidellaquila** [comment]
Hi @christinedraper 

I just pushed a fix for the NPE, the fix will be released with 2.2.18.

Please consider that in v 2.2.x this query won't use the index (the query executor cannot figure out that the subquery can be executed only once, so it tries to execute it record by record), this issue is already fixed in develop branch, so in v 3.0 we will have optimized indexed execution.

Thanks

Luigi

**Comment by christinedraper** [comment]
Thanks!
